test/functional/page_objects/discover_page.ts

```diff
@@ -532,7 +532,12 @@ export class DiscoverPageObject extends FtrService {
     await this.retry.waitFor('the button before pressing it', async () => {
       return await this.testSubjects.exists('discoverNoResultsViewAllMatches');
     });
-    return await this.testSubjects.click('discoverNoResultsViewAllMatches');
+    await this.testSubjects.click('discoverNoResultsViewAllMatches');
+    await this.retry.waitFor('the button to disappear', async () => {
+      return !(await this.testSubjects.exists('discoverNoResultsViewAllMatches'));
+    });
```

Comment:
I believe this approach functions the same as the original approach since it's similar to these lines that already existed, which also tried just waiting until the UI updates: https://github.com/elastic/kibana/blob/99a1106999dc427765cf47dcda79646f453f42b1/x-pack/test_serverless/functional/test_suites/common/discover/group1/_discover.ts#L179-L182